spyder/plugins/editor/plugin.py

```diff
@@ -2959,39 +2973,6 @@ def setup_open_files(self):
             self.__load_temp_file()
         self.set_create_new_file_if_empty(True)
 
-    def reorder_filenames(self, filenames):

```

Comment:
This method seems to cover more things than what your current code does (e.g. it seems to handle splitted editor stacks). Please explain why you're removing it.

Reply:
The **only** thing this method does, is to load the `splitsettings`, take the filename that had focus last in the first `EditorStack` (`index = 0`), move it to position 0, and save the modified `splitsettings`. 

The settings for the additional splitted editor stacks (`index > 0`) are not modified by this.

This PR proposes to preserve the file order that was saved when Spyder was last closed and to stop reordering files during the setup. Therefore, with the proposed changes, this method is not used anywhere else in the code. I can keep it however if you prefer in case it is needed for something else at some point.

Reply:
But does focus jumps from one file to another while the Editor is opening them? That's my main concern here.

Reply:
No it does not. I'll post an animation to show you what it is doing, one sec.

Reply:
Ok, that's good. Preserving file order and loading the last file with focus first is ice, UX-wise.

@CAM-Gerlach, what do you think?

Reply:
The Outline Explorer is jumping around though, but it was doing that before.At least with this PR, it ends up showing the tree of the right file at the end of the setup. I can fix this in another PR if you think it should.

Reply:
Yeah, that's fine and does look a little nicer at startup as well, although since its only transient its not the biggest concern for me personally.

Of course, the outline explorer is still ordered alphabetically, rather than the order of the tabs in the Editor. Presumably out of scope for now, but if/when you make that change, if would be nice to have the option to either alphabetize it or make it follow the order of the editor, since both could potentially be useful depending on one's workflow.

Also, to note, the outline explorer only show Python files, so it isn't a true file switcher for more general cases. Perhaps, as a longer-term idea, we could have an option to show them (just the files without the outlines, of course).

Reply:
> Yeah, that's fine and does look a little nicer at startup as well, although since its only transient its not the biggest concern for me personally.

The most important part of this change is not the transient effect, but the fact that we do not need to reorder the tabs during setup.